src/rpc/mining.cpp

```diff
@@ -139,18 +139,23 @@ UniValue generate(const UniValue& params, bool fHelp)
     const int nGenerate = params[0].get_int();
     int nHeightEnd = 0;
     int nHeight = 0;
-    CReserveKey reservekey(pwalletMain);
 
     {   // Don't keep cs_main locked
         LOCK(cs_main);
         nHeight = chainActive.Height();
         nHeightEnd = nHeight + nGenerate;
     }
-    unsigned int nExtraNonce = 0;
-    UniValue blockHashes(UniValue::VARR);
 
-    bool fPoS = false;
     const int last_pow_block = Params().LAST_POW_BLOCK();
+    if (nHeight > last_pow_block) {
```

Comment:
this should be `>=` I think, as, when the chain height is `last_pow_block`, the wallet is trying to create the first PoS block (and thus has to be unlocked).

Might make sense to define `fPoS` up here and use it also outside the `while`.

Reply:
yep, done.